### PR TITLE
Feature/project restructure

### PR DESCRIPTION
**Pull request by earlgaspard**
- Project restructure to conform to other BR open source projects
- Added swiftlint
- Added a changelog

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,28 +2,26 @@
 # * http://www.objc.io/issue-6/travis-ci.html
 # * https://github.com/supermarin/xcpretty#usage
 
-osx_image: xcode9.2
+osx_image: xcode9.3
```
Comment (wmcginty) [inline .travis.yml:5]: We should look at starting to migrate to 9.4 images where possible/needed

**Review by wmcginty — APPROVED** [review]
Looks good, @earlgaspard

File: .travis.yml (hunk 1 above)
Comment (earlgaspard) [reply .travis.yml:5]: I attempted to do so but it didn't work for 9.4. could be they haven't set it up yet. is this the source of truth?  https://docs.travis-ci.com/user/reference/osx/

Comment (wmcginty) [reply .travis.yml:5]: As far as I know, yes: https://github.com/travis-ci/travis-ci/issues/9689

**Pull request merged**